src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -255,4 +222,92 @@ private function getSortClauses(
             return []; // rely on storage engine default sorting
         }
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo[] $contentInfoList
+     *
+     * @return \EzSystems\EzPlatformAdminUi\REST\Value\ContentTree\Node
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    private function buildNode(
+        Location $location,
+        ?LoadSubtreeRequestNode $loadSubtreeRequestNode = null,
+        bool $loadChildren = false,
+        int $depth = 0,
+        ?string $sortClause = null,
+        string $sortOrder = Query::SORT_ASC,
+        ArrayObject $contentInfoList = null
+    ): Node {
+        $contentInfo = $location->getContentInfo();
+        $contentId = $location->contentId;
+        if (!isset($this->contentInfoList[$contentId])) {
+            $contentInfoList[$contentId] = $contentInfo;
```

Comment:
`$contentInfoList` might not be initialized at this point.

You probably want to add...
```php
$contentInfoList = $contentInfoList ?? new ArrayObject();
```
...at the beginning of this method.